util/topgen.py

```diff
@@ -367,102 +450,131 @@ def generate_pinmux(top: Dict[str, object], out_path: Path) -> None:
         "scan_role_pkg_vlnv": f"lowrisc:systems:top_{topname}_scan_role_pkg",
     }
 
-    ipgen_render("pinmux", topname, params, out_path)
 
+def generate_pinmux(top: Dict[str, object], module: Dict[str, object],
+                    out_path: Path) -> None:
+    log.info("Generating pinmux with ipgen")
+    params = _get_pinmux_params(top)
+    generate_ipgen(top, module, params, out_path)
 
-# generate clkmgr with ipgen
-def generate_clkmgr(topcfg: Dict[str, object], out_path: Path) -> None:
-    log.info("Generating clkmgr with ipgen")
-    topname = topcfg["name"]
 
-    clocks = topcfg["clocks"]
+def _get_clkmgr_params(top: Dict[str, object]) -> Dict[str, object]:
+    """Gets parameters for clkmgr ipgen from top config."""
+    clocks = top["clocks"]
     assert isinstance(clocks, Clocks)
 
     typed_clocks = clocks.typed_clocks()
     hint_names = typed_clocks.hint_names()
 
     typed_clks = OrderedDict({
-        ty: {nm: {"src_name": sig.src.name, "endpoint_ip": sig.endpoints[0][0]}
-             for nm, sig in mp.items() if isinstance(sig, ClockSignal)}
-        for ty, mp in typed_clocks._asdict().items() if isinstance(mp, dict)})
+        ty: {
+            nm: {
+                "src_name": sig.src.name,
+                "endpoint_ip": sig.endpoints[0][0]
+            }
+            for nm, sig in mp.items() if isinstance(sig, ClockSignal)
+        }
+        for ty, mp in typed_clocks._asdict().items() if isinstance(mp, dict)
+    })
 
     # Will connect to alert_handler
-    with_alert_handler = lib.find_module(topcfg['module'], 'alert_handler') is not None
-
-    params = {
-        "src_clks": OrderedDict({
-            name: vars(obj) for name, obj in clocks.srcs.items()}),
-        "derived_clks": OrderedDict({
-            name: vars(obj) for name, obj in clocks.derived_srcs.items()}),
-        "typed_clocks": OrderedDict({
-            ty: d for ty, d in typed_clks.items() if d}),
-        "hint_names": hint_names,
-        "parent_child_clks": typed_clocks.parent_child_clks,
-        "exported_clks": topcfg["exported_clks"],
-        "number_of_clock_groups": len(clocks.groups),
-        "with_alert_handler": with_alert_handler,
+    with_alert_handler = lib.find_module(top['module'],
+                                         'alert_handler') is not None
+
+    topname = top["name"]
+    return {
+        "src_clks":
+        OrderedDict({name: vars(obj)
+                     for name, obj in clocks.srcs.items()}),
```

Comment:
Is that the correct indentation for style / readability? When keys and values are on separate lines, the values don't get indented?

(just a question -- it feels a bit hard to read for me)